doc/design/bucketpath.rst

```diff
@@ -190,44 +197,97 @@ Pathlike
 
         def read(chunk_size: int = 8192) -> Iterable[ByteString]:
             """
-            TBD
+            Read the content of a the file behind this path.
+
+            Only works for pathslike objects which return True for `is_file()`.
+
+            Args:
+                chunk_size: which will be yieled by the iterator.
+
+            Returns:
+                Returns an iterator which can be used to read the contents of the path in chunks.
+
+            Raises:
+                NotAFileError: if the pathlike object does not point to a file.
+                FileNotFoundError: If the file does not exist.
             """
 
         def write(data: ByteString | BinaryIO | Iterable[ByteString]):
             """
-            TBD
+            Writes data to a this path. 
+
+            After successfully writing to this path `exists` will yield true for this path.
+            If the file already existed it will be overwritten.
+
+            Args:
+                data: which shall be writen to the path.
+
+            Raises:
+                NotAFileError: if the pathlike object is not a file path.
             """
 
         def rm():
             """
-            If exists() and is_file yields true for this path, the path will be deleted,
-            otherwise exception will be thrown.
+            Remove this file.
+
+            Note:
+                If `exists()` and is_file yields true for this path, the path will be deleted,
+                otherwise exception will be thrown.
+
+            Raises:
+                FileNotFoundError: If the file does not exist.
             """
 
         def rmdir(recursive: bool = False):
             """
-            TBD
+            Removes this directory.
+
+            Note: In order to stay close to the pathlib based `rmdir` the default won't delete
+                  non empty direcotries.
+
+            Args:
+                recursive: if true the entrie directory and it's entire contents (files and subdirs)
```

Comment:
```suggestion
                recursive: if true the directory itself and it's entire contents (files and subdirs)
```

Reply:
In this case it should be 'its'.

Reply:
Thx @ both, in that case I will manually adjust ;)